statics/workflows/pkt-validation.yaml

```diff
@@ -0,0 +1,451 @@
+  - Name: source
+    Description: Source Nodes (Select Source Node)
+    Type: node
+  - Name: destination
+    Description: Destination Node (Select Destination Node)
+    Type: node
+  - Name: capture_port
+    Description: Capture Port (BPF Filter)
+    Type: string
+  - Name: Flow Application Type
+    Description: Flow Application Type (Protocol to Filter Flows)
+    Type: choice
+    Default: All
+    Values:
+      - Description: "Protocol : ICMP"
+        Value: ICMP
+      - Description: "Protocol : TCP"
+        Value: TCP
+      - Description: "Protocol : UDP"
+        Value: UDP
+      - Description: "All Protocols"
+        Value: All
+  - Name: IP Version
+    Description: IP version (IPV4/IPV6)
+    Type: choice
+    Default: All
+    Values:
+      - Description: "IP Version : IPV4"
+        Value: IPV4
+      - Description: "IP Version : IPV6"
+        value: IPV6
+      - Description: "IP Version : Both"
+        value: All
+  - Name: ID/Src-Dst Port
+    Description: Enter ICMP-IDs Like (ID1,ID2,ID3,....), Src-Dst Port No. Like (srcport1-dstport1,srcport2-dstport2,srcport3-,-dstport4...) or both like (ID1,ID2,...,srcport1-dstport1,...)
+    Type: string
+  - Name: Flow Filtering Parameters
+    Description: Enter Flow Filtering Parameters as a String like (Key1=Value1,Key2=Value2,....)
+    Type: string
+  - Name: Time Duration for Capture the flows
+    Description: Time Duration to capture the flows in seconds
+    Type: integer
+Source: |
+  function PacketForwardingValidation(src, dst, bpf, app, ipVersion, idString, filterString, interval) {
+
+    var parseParams = function(idString, filterString) {
+      var icmpID = [];
+      var ports = [];
+      var srcPort = [];
+      var dstPort = [];
+      var filters = [];
+      var a, b = 0;
+      var str1 = idString.split(",");
+      for (var i = 0; i < str1.length; i++) {
+        if (str1[i].indexOf("-") >= 0) {
+          ports[a] = str1[i].split("-");
+          srcPort[a] = parseInt(ports[a][0]);
+          dstPort[a] = parseInt(ports[a][1]);
+          a++;
+        } else {
+          icmpID[b] = parseInt(str1[i]);
+          b++;
+        }
+      }
+      var str2 = filterString.split(",");
+      for (var i = 0; i < str2.length; i++) {
+        filters[i] = str2[i].split("=");
+      }
+      var parseparam = {"icmpID" : icmpID, "srcPort" : srcPort, "dstPort" : dstPort, "filters" : filters}
+      return parseparam;
+    }
+
+    var getFlows = function(src, dst, app, ipVersion, interval, parseparam) {
+      var icmpID = parseparam["icmpID"];
+      var srcPort = parseparam["srcPort"];
+      var dstPort = parseparam["dstPort"];
+      var filters = parseparam["filters"];
+      var nodes = {};
+      var paths = client.G.V().Has('TID', src).ShortestPathTo(Metadata('TID', dst), Metadata('RelationType', 'layer2')).result();
+      var path = paths[0]
+      var pathNodes = [];
+      var aa = 0;
+      for (i in path) {
+       pathNodes[aa] = path[i];
```

Comment:
you want to clone an array here? if so you can use var pathNodes = path.slice().

so you can remove the loop and the var aa